File: android/src/main/java-v10/com/mapbox/rnmbx/components/annotation/RNMBXCalloutManager.java
package com.mapbox.rnmbx.components.annotation;

import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;

public class RNMBXCalloutManager extends ViewGroupManager<RNMBXCallout> {
    public static final String REACT_CLASS = "RNMBXCallout";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected RNMBXCallout createViewInstance(ThemedReactContext reactContext) {
        return new RNMBXCallout(reactContext);
    }
}

File: android/src/main/java/com/rnmapbox/rnmbx/components/annotation/RNMBXCalloutManager.kt
package com.rnmapbox.rnmbx.components.annotation

import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.ViewManagerDelegate
import com.facebook.react.viewmanagers.RNMBXCalloutManagerDelegate
import com.facebook.react.viewmanagers.RNMBXCalloutManagerInterface

class RNMBXCalloutManager : ViewGroupManager<RNMBXCallout>(),
    RNMBXCalloutManagerInterface<RNMBXCallout> {

    private val mDelegate: ViewManagerDelegate<RNMBXCallout>

    init {
        mDelegate = RNMBXCalloutManagerDelegate(this)
    }

    override fun getDelegate(): ViewManagerDelegate<RNMBXCallout> {
        return mDelegate
    }

    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createViewInstance(reactContext: ThemedReactContext): RNMBXCallout {
        return RNMBXCallout(reactContext)
    }

    companion object {
        const val REACT_CLASS = "RNMBXCallout"
    }
}